source/common/router/retry_state_impl.cc

```diff
@@ -157,6 +153,10 @@ RetryStatus RetryStateImpl::shouldRetry(const Http::HeaderMap* response_headers,
     return RetryStatus::NoOverflow;
   }
 
+  if (!runtime_.snapshot().featureEnabled("upstream.use_retry", 100)) {
```

Comment:
This was moved for perf reasons only. No point in doing lookup unless we are actually going to retry.